asciidoctorj-api/src/main/java/org/asciidoctor/ast/Row.java

```diff
@@ -4,6 +4,6 @@
 
 public interface Row {
 
-    public List<Cell> getCells();
+    List<? extends Cell> getCells();
```

Comment:
I think this is not going to work.
While the getters on StructuredDocument etc are read-only collections, these are meant to be mutated by extensions.
With this change extensions will no longer be able to add columns, headers, footers, and rows.
I guess we have to figure out how to make the Groovy based tests spot such things.
As these tests are not compiled static at the moment Groovy just adds values to these members, but Java wouldn't allow it.

So using wildcards for these 5 interfaces should be fine.
- Author
- ContentPart
- DocumentHeader
- RevisionInfo
- StructuredNode
Collections returned by all other AST interfaces should stay the same because they are intended to be mutated by extensions.

See for example https://github.com/asciidoctor/asciidoctorj/blob/5b8018792c787f59df8ee099d5f8393f68e7591b/asciidoctorj-core/src/test/groovy/org/asciidoctor/extension/GithubContributorsBlockMacro.groovy#L52

Reply:
I guess you mean `StructuredDocument` (`StructuredNode` does not exist)

Reply:
Of course, you're right.
This should've been StructuredDocument.